### DISCUSSION TITLE
Milvus: BulkInsert using files in Java

### DISCUSSION THREAD

**Discussion by UshaPravinkumar** [milvus-io/milvus#30585] [Q&A and General discussion] [answered]
Hello team, 
I followed the official document to use Milvus' Java SDK to support bulk insertions. 
https://milvus.io/docs/bulk_insert.md#Insert-entities-from-files

I'm using a bucket file for the bulk insert operation, not a local directory file. I'm new to python and most of the documents use python.
I got the result like ImportFailed with Failed Reason:failed to get file size of 'BUCKET_NAME/FILE_NAME.json', error:The specified key does not exist.

milvus-sdk-java : 2.3.4 version
OS : Windows 11

**Comment by UshaPravinkumar** [comment]
I uploaded a new file object using MinioClient to the bucket. When I process the same uploaded object, I get the same error.

**Reply by yhmo** [reply]
The file is not in correct path.

Assume the milvus is using S3 bucket named "a-bucket".
You uploaded the 1.json file to root path of "a-bucket". You call the api in this way:
`utility.do_bulk_insert(collection_name="xxx", files=["1.json"])`

If you uploaded the 1.json file to /aaa/bbb/1.json, you call the api in this way:
`utility.do_bulk_insert(collection_name="xxx", files=["/aaa/bbb/1.json"])`

**The path should not include the bucket name.**

**Reply by UshaPravinkumar** [reply]
The file (ex: 1.json) is placed under a bucket "a-bucket". I tried the below codes.
Initially tried with addFile option and then tried withFiles. But both the methods returns the same error "error:The specified key does not exist."

milvusServiceClient.bulkInsert(BulkInsertParam.newBuilder().withCollectionName(collectionName).addFile("1.json").build());

milvusServiceClient.bulkInsert(BulkInsertParam.newBuilder().withCollectionName(collectionName).withFiles(Arrays.asList("1.json")).build());

Also tried to include bucket name in the file name "a-bucket/1.json", but same error.
Do I need to check/change any configuration setup?

**Reply by yhmo** [reply, marked as answer]
The bucket must be the same bucket which the milvus server is using.

**Comment by amaypatil** [comment]
@yhmo @xiaofan-luan Is there a way for bulk insert to have source=different s3 bucket/objectstore path than the one used by the Milvus cluster

**Reply by yhmo** [reply]
Milvus can only access its s3 bucket to import file. If your files are in another bucket or another object store, you need to copy the files into the milvus s3 bucket by some other tools or object store clients.
